cmd/lotus-miner/actor.go

```diff
@@ -206,10 +211,307 @@ var actorSetPeeridCmd = &cli.Command{
+		if err != nil {
+			return err
+		}
+
+		value, err := types.ParseFIL(cctx.Args().Get(2))
+		if err != nil {
+			return err
+		}
+
+		available, err := api.StateMinerAvailableBalance(ctx, msig, types.EmptyTSK)
+		if err != nil {
+			return err
+		}
+
+		var method uint64
+		var params []byte
+		if cctx.Args().Len() == 5 {
+			m, err := strconv.ParseUint(cctx.Args().Get(3), 10, 64)
+			if err != nil {
+				return err
+			}
+			method = m
+
+			// get amount
+			amount := available
+			if cctx.Args().Present() {
+				f, err := types.ParseFIL(cctx.Args().Get(4))
+				if err != nil {
+					return xerrors.Errorf("parsing 'amount' argument: %w", err)
+				}
+
+				amount = abi.TokenAmount(f)
+
+				if amount.GreaterThan(available) {
+					return xerrors.Errorf("can't withdraw more funds than available; requested: %s; available: %s", types.FIL(amount), types.FIL(available))
+				}
+			}
+
+			params, err = actors.SerializeParams(&miner.WithdrawBalanceParams{
+				AmountRequested: amount, // Default to attempting to withdraw all the extra funds in the miner actor
+			})
+			if err != nil {
+				return err
+			}
+
+		}
+
+		var from address.Address
+		if cctx.IsSet("from") {
+			f, err := address.NewFromString(cctx.String("from"))
+			if err != nil {
+				return err
+			}
+			from = f
+		} else {
+			defaddr, err := api.WalletDefaultAddress(ctx)
+			if err != nil {
+				return err
+			}
+			from = defaddr
+		}
+
+		act, err := api.StateGetActor(ctx, msig, types.EmptyTSK)
+		if err != nil {
+			return fmt.Errorf("failed to look up multisig %s: %w", msig, err)
+		}
+
+		if !builtin.IsMultisigActor(act.Code) {
+			return fmt.Errorf("actor %s is not a multisig actor", msig)
+		}
+
+		proto, err := api.MsigPropose(ctx, msig, dest, types.BigInt(value), from, method, params)
+		if err != nil {
+			return err
+		}
+
+		sm, err := lcli.InteractiveSend(ctx, cctx, srv, proto)
+		if err != nil {
+			return err
+		}
+
+		msgCid := sm.Cid()
+
+		fmt.Println("sent proposal in message: ", msgCid)
+
+		wait, err := api.StateWaitMsg(ctx, msgCid, uint64(cctx.Int("confidence")), build.Finality, true)
+		if err != nil {
+			return err
+		}
+
+		if wait.Receipt.ExitCode != 0 {
+			return fmt.Errorf("proposal returned exit %d", wait.Receipt.ExitCode)
+		}
+
+		var retval msig2.ProposeReturn
+		if err := retval.UnmarshalCBOR(bytes.NewReader(wait.Receipt.Return)); err != nil {
+			return fmt.Errorf("failed to unmarshal propose return value: %w", err)
+		}
+
+		fmt.Printf("Transaction ID: %d\n", retval.TxnID)
+		if retval.Applied {
+			fmt.Printf("Transaction was executed during propose\n")
+			fmt.Printf("Exit Code: %d\n", retval.Code)
+			fmt.Printf("Return Value: %x\n", retval.Ret)
+		}
+
+		return nil
+	},
+}
+var actorWithdrawMsigApproveCmd = &cli.Command{
+	Name:      "approve",
+	Usage:     "Approve a multisig message",
+	ArgsUsage: "<multisigAddress messageId> [proposerAddress destination value [methodId methodParams]]",
```

Comment:
This function should take 2 required args `multisigAddress`, `messageId`, and 3 optional args `proposerAddress` `destinationAddress` , `withdrawlAmount`